MmSupervisorPkg/Core/MmSupervisorCore.c

```diff
@@ -798,6 +799,133 @@ DiscoverStandaloneMmDriversInFvHobs (
   return EFI_SUCCESS;
 }
 
+/**
+  Routine for initializing policy data provided by firmware.
+
+  @retval EFI_SUCCESS           The handler for the processor interrupt was successfully installed or uninstalled.
+  @retval Errors                The supervisor is unable to locate or protect the policy from firmware.
+
+**/
+EFI_STATUS
+InitializePolicy (
+  VOID
+  )
+{
+  EFI_STATUS           Status;
+  EFI_FFS_FILE_HEADER  *FileHeader;
+  VOID                 *SectionData;
+  UINTN                SectionDataSize;
+  UINTN                PolicySize;
+
+  FirmwarePolicy = NULL;
+
+  //
+  // First try to find the policy file based on the GUID specified.
+  //
+  FileHeader = NULL;
+  do {
+    Status =  FfsFindNextFile (
+                EFI_FV_FILETYPE_FREEFORM,
+                (EFI_FIRMWARE_VOLUME_HEADER *)gMmCorePrivate->StandaloneBfvAddress,
+                &FileHeader
+                );
+    if (EFI_ERROR (Status)) {
+      DEBUG ((
+        DEBUG_ERROR,
+        "[%a] Failed to locate firmware policy file from given FV - %r\n",
+        __FUNCTION__,
+        Status
+        ));
+      break;
+    }
+
+    if (!CompareGuid (&FileHeader->Name, &gMmSupervisorPolicyFileGuid)) {
+      continue;
+    }
+
+    DEBUG ((
+      DEBUG_INFO,
+      "[%a] Discovered policy file in FV at 0x%p.\n",
+      __FUNCTION__,
+      FileHeader
+      ));
+
+    Status = FfsFindSectionData (
+               EFI_SECTION_RAW,
+               FileHeader,
+               &SectionData,
+               &SectionDataSize
+               );
+    if (EFI_ERROR (Status)) {
+      DEBUG ((
+        DEBUG_ERROR,
+        "[%a] Failed to find raw section from discovered policy file - %r\n",
+        __FUNCTION__,
+        Status
+        ));
+      break;
+    }
+
+    PolicySize = ((SMM_SUPV_SECURE_POLICY_DATA_V1_0 *)SectionData)->Size;
+    if (PolicySize > SectionDataSize) {

```

Comment:
Why do we only care if the Policy Size is larger than the SectionDataSize?  I just want more information on what we're checking here.

Reply:
This means that the policy binary we carried in the FV is smaller than the policy data indicates. This normally means that the policy data is incorrect, but could also mean that the system is under attack.

Reply:
I understand why that might be a problem.  I was more wondering about why it's okay not to check if the PolicySize is smaller than SectionDataSize.  Wouldn't that also mean there is an error or there was tampering?

Reply:
It might mean that there some bytes that we do not understand. But if that is the case, the security check below will call out what bytes are unexpected.